src/tools.cc

```diff
@@ -136,17 +137,89 @@ String trim_string(String str)
         }
     }
     if (i >= len)
-        return _("");
+        return "";
     for (i = len - 1; i >= start; i--) {
         if (!strchr(WHITE_SPACE, buf[i])) {
             end = i + 1;
             break;
         }
     }
-    return str.substring(start, end - start);
+    return str.substr(start, end - start);
+}
+
+bool startswith_string(std::string str, std::string check)
+{
+    return str.rfind(check) == 0;
```

Comment:
Shouldnt this be `rfind( check, 0)`? Otherwise we search the whole string?

I feel like all the utility methods could lose the `_string`, the typing makes it pretty obvious, and the code tidier

Reply:
good catch, i have fixed this

Reply:
- renaming tolower_string to tolower leads to problem as it clashes with existing symbols (same for stoi_string)

Maybe it makes sense to introduce a util/string.cc/h modules containing all string util functions. There we could stream line the functions like:
- X_str or str_X (e.g. X == tolower, ....)
- or you have a better idea

Reply:
or static class or namespace :-)

Reply:
Yes, to any of those :)